tools/infrasys/src/root.rs

```diff
@@ -0,0 +1,195 @@
+        trace!("tuftool split args: {:#?}", args);
+
+        let status = Command::new("tuftool")
+            .args(args)
+            .env("AWS_REGION", $region)
+            .status()
+            .context(error::TuftoolSpawn)?;
+
+        ensure!(status.success(), error::TuftoolResult {
+            command: arg_str,
+            code: status.code().map(|i| i.to_string()).unwrap_or_else(|| "<unknown>".to_string())
+        });
+    }
+}
+
+pub fn check_root(root_role_path: &Path) -> Result<()> {
+    ensure!(!root_role_path.is_file(), {
+        warn!("Cowardly refusing to overwrite the existing root.json at {}. Please manually delete it and run again.", root_role_path.display());
+        error::FileExists {
+            path: root_role_path,
+        }
+    });
+    Ok(())
+}
+
+/// Creates the directory where root.json will live and creates root.json itself according to details specified in root-role-path
+pub fn create_root(root_role_path: &Path) -> Result<()> {
+    // Make /roles and /keys directories, if they don't exist, so we can write generated files.
+    let role_dir = root_role_path.parent().context(error::InvalidPath {
+        path: root_role_path,
+        thing: "root role",
+    })?;
+    fs::create_dir_all(role_dir).context(error::Mkdir { path: role_dir })?;
+    // Initialize root
+    tuftool!(
+        Region::default().name(),
+        "root init '{}'",
+        root_role_path.display()
+    );
+    tuftool!(
+        Region::default().name(),
+        // TODO: expose expiration date as a configurable parameter
+        "root expire '{}' 'in 52 weeks'",
+        root_role_path.display()
+    );
+    Ok(())
+}
+
+/// Adds keys to root.json according to key type  
+pub fn add_keys(
+    signing_key_config: &mut SigningKeyConfig,
+    role: &KeyRole,
+    threshold: &str,
+    filepath: &str,
+) -> Result<()> {
+    match signing_key_config {
+        SigningKeyConfig::file { .. } => (),
+        SigningKeyConfig::kms { key_id, config, .. } => add_keys_kms(
+            &config
+                .as_ref()
+                .context(error::MissingConfig {
+                    missing: "config field for a kms key",
+                })?
+                .available_keys,
+            role,
+            threshold,
+            filepath,
+            key_id,
+        )?,
+        SigningKeyConfig::ssm { .. } => (),
+    }
+    Ok(())
+}
+
+/// Adds KMSKeys to root.json given root or publication type
+/// Input: available-keys (keys to sign with), role (root or publication), threshold for role, filepath for root.JSON,
+/// mutable key_id
+/// Output: in-place edit of root.json and key_id with a valid publication key
+/// (If key-id is populated, it will not change. Otherwise, it will be populated with a key-id of an available key)
+fn add_keys_kms(
+    available_keys: &HashMap<String, String>,
+    role: &KeyRole,
+    threshold: &str,
```

Comment:
Get this parsed as part of `ValidRepoInfo` and pass it everywhere as `NonZeroUsize`. See comment at `ValidRepoInfo` as well.

```suggestion
    threshold: NonZeroUsize,
```